### PR TITLE
Screenshot tools and tasks

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

This initial PR basically re-writes the existing screenshot tool (that no longer worked) and moves it into our `__task__` and `make` architecture.

Currently it only screenshots the three edition fronts, the intention being to extend this to a useful set of paths that include all content types. 

Long term - @MatthewJWalls is working on getting PR builds up into instances which we can then screenshot. We can also look at cross-browser screenshot or visual regression services and spit this into PRs. Grouping page types and focussing in on particular sections of pages may also be useful.

## What is the value of this and can you measure success?

- Allows a simple visual check to catch oddities with changes
- Lays some of the groundwork for useful PR build screenshotting

## Does this affect other platforms - Amp, Apps, etc?

A visual check of AMP will certainly be useful in the future.

## Screenshots

screenies

How do you screenshot screenshots.. wut

image

Bonus happy dog.

teeth

## Request for comment

@guardian/dotcom-platform

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
Oooh super interesting @gtrufitt 

We have a use case where this could come in handy, especially if we could pass an optional path as argument :-)

Might be worth discussing with @guardian/commercial-dev

**Comment by michaelwmcnamara** [comment]
<iframe src="//giphy.com/embed/l0O7P6qdCa1AKJRAY" width="480" height="351" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="http://giphy.com/gifs/studio-90s-l0O7P6qdCa1AKJRAY">via GIPHY</a></p>

**Comment by michaelwmcnamara** [comment]
yeah

File: tools/__tasks__/screenshot/screenshot.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ *  - Generate nicer viewing output
+ *  - Generate copy paste into github
+ *  - Screenshot 'components' by classname (captureSelector)
+ */
+
+const {paths, breakpoints, host, screenshotsDir, environment} = require('./config');
+
+function takeScreenshots() {
+    const taskArray =
+        paths.map((path) => {
```
Comment (sndrs) [inline tools/__tasks__/screenshot/screenshot.js]: why not just return this `.map`?

Comment (sndrs) [inline tools/__tasks__/screenshot/screenshot.js]: i.e. something like 
```js
task: paths.map(path => 
    Object.keys(breakpoints).map(breakpointName => {
        ...
    })
)
```

**Review by sndrs — APPROVED** [review]
couple of comments but not blockers - nice!

**Comment by gtrufitt** [comment]
image

@SiAdcock @ScottPainterGNM used em

**Comment by ScottPainterGNM** [comment]
I would be mad but this is a cause I'm very much behind

**Pull request merged**